File: 66-plus-one/plus-one.py
class Solution:
    # def rec(self, num: int) -> int:
    def plusOne(self, digits: List[int]) -> List[int]:
        #[1,2,3,9] - if 9 - push one index back 
        for i in range(len(digits)-1,-1,-1):
            if digits[i] == 9:
                digits[i] = 0
                if i == 0:
                    digits = [1] + digits
            else:
                digits[i]+=1
                break
        return digits
        

File: 66-plus-one/plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int s = digits.size();
        if(digits[s-1] >= 0 && digits[s-1] <= 8) {
            digits[s-1] = digits[s-1] + 1;
        } else {
            for(int i = s - 1; i >= 0; i--) {
                if(digits[i] == 9) {
                    digits[i] = 0;
                    if(i == 0) {
                        digits.insert(digits.begin(), 1);
                    }
                } else {
                    digits[i] = digits[i] + 1;
                    break;
                }
            }
        }
        return digits;
    }
};